File: app/src/main/java/com/bridou_n/beaconscanner/dagger/BluetoothModule.java
package com.bridou_n.beaconscanner.dagger;

import android.bluetooth.BluetoothAdapter;
import android.content.Context;
import android.support.annotation.Nullable;

import org.altbeacon.beacon.BeaconManager;
import org.altbeacon.beacon.BeaconParser;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by bridou_n on 05/10/2016.
 */

@Module
public class BluetoothModule {

    @Nullable @Provides @PerActivity
    public BluetoothAdapter providesBluetoothAdapter() {
        return BluetoothAdapter.getDefaultAdapter();
    }

    @Provides @PerActivity
    public BeaconManager providesBeaconManager(Context ctx) {
        BeaconManager instance = BeaconManager.getInstanceForApplication(ctx);

        // BeaconManager setup
        instance.getBeaconParsers().add(new BeaconParser().setBeaconLayout("m:0-3=4c000215,i:4-19,i:20-21,i:22-23,p:24-24"));
        // Detect the main identifier (UID) frame:
        instance.getBeaconParsers().add(new BeaconParser().setBeaconLayout("s:0-1=feaa,m:2-2=00,p:3-3:-41,i:4-13,i:14-19"));
        // Detect the telemetry (TLM) frame:
        instance.getBeaconParsers().add(new BeaconParser().setBeaconLayout("x,s:0-1=feaa,m:2-2=20,d:3-3,d:4-5,d:6-7,d:8-11,d:12-15"));
        // Detect the URL frame:
        instance.getBeaconParsers().add(new BeaconParser().setBeaconLayout("s:0-1=feaa,m:2-2=10,p:3-3:-41,i:4-21v"));

        return instance;
    }
}

File: app/src/main/java/com/bridou_n/beaconscanner/dagger/modules/BluetoothModule.kt
package com.bridou_n.beaconscanner.dagger.modules

import android.bluetooth.BluetoothAdapter
import android.content.Context

import com.bridou_n.beaconscanner.dagger.PerActivity
import com.bridou_n.beaconscanner.utils.PreferencesHelper

import org.altbeacon.beacon.BeaconManager
import org.altbeacon.beacon.BeaconParser

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import org.altbeacon.beacon.Beacon

/**
 * Created by bridou_n on 05/10/2016.
 */

@Module
class BluetoothModule {

    companion object {
        const val RUUVI_LAYOUT = "m:0-2=0499,i:4-19,i:20-21,i:22-23,p:24-24" // TBD
        const val IBEACON_LAYOUT = "m:0-3=4c000215,i:4-19,i:20-21,i:22-23,p:24-24"
        const val ALTBEACON_LAYOUT = BeaconParser.ALTBEACON_LAYOUT
        const val EDDYSTONE_UID_LAYOUT = BeaconParser.EDDYSTONE_UID_LAYOUT
        const val EDDYSTONE_URL_LAYOUT = BeaconParser.EDDYSTONE_URL_LAYOUT
        const val EDDYSTONE_TLM_LAYOUT = BeaconParser.EDDYSTONE_TLM_LAYOUT
    }

    @Provides
    @PerActivity
    fun providesBluetoothAdapter(): BluetoothAdapter? {
        return BluetoothAdapter.getDefaultAdapter()
    }

    @Provides // Not a Singleton
    fun providesBeaconManager(ctx: Context, prefs: PreferencesHelper): BeaconManager {
        val instance = BeaconManager.getInstanceForApplication(ctx)

        // Sets the delay between each scans according to the settings
        instance.foregroundBetweenScanPeriod = prefs.getScanDelay()

        // Add all the beacon types we want to discover
        instance.beaconParsers.add(BeaconParser().setBeaconLayout(IBEACON_LAYOUT))
        instance.beaconParsers.add(BeaconParser().setBeaconLayout(EDDYSTONE_UID_LAYOUT))
        instance.beaconParsers.add(BeaconParser().setBeaconLayout(EDDYSTONE_URL_LAYOUT))
        instance.beaconParsers.add(BeaconParser().setBeaconLayout(EDDYSTONE_TLM_LAYOUT))

        return instance
    }
}
